app/components/post_draft/send_handler/send_handler.tsx

```diff
@@ -0,0 +1,302 @@
+import {preventDoubleTap} from '@utils/tap';
+import {confirmOutOfOfficeDisabled} from '@utils/user';
+
+import CursorPositionHandler from '../cursor_position_handler';
+
+type Props = {
+    testID?: string;
+    channelId: string;
+    rootId: string;
+
+    // From database
+    currentUserId: string;
+    enableConfirmNotificationsToChannel?: boolean;
+    isTimezoneEnabled: boolean;
+    maxMessageLength: number;
+    membersCount?: number;
+    useChannelMentions: boolean;
+    userIsOutOfOffice: boolean;
+    useGroupMentions: boolean;
+    groupsWithAllowReference: Group[];
+
+    // DRAFT Handler
+    value: string;
+    files: FileInfo[];
+    clearDraft: () => void;
+    updateValue: (message: string) => void;
+    addFiles: (file: FileInfo[]) => void;
+    removeFiles: (file: FileInfo) => void;
+    retryFileUpload: (file: FileInfo) => void;
+    uploadFileError: React.ReactNode;
+}
+
+export default function SendHandler({
+    testID,
+    channelId,
+    currentUserId,
+    enableConfirmNotificationsToChannel,
+    files,
+    isTimezoneEnabled,
+    maxMessageLength,
+    membersCount = 0,
+    rootId,
+    useChannelMentions,
+    userIsOutOfOffice,
+    value,
+    useGroupMentions,
+    groupsWithAllowReference,
+    clearDraft,
+    updateValue,
+    addFiles,
+    removeFiles,
+    retryFileUpload,
+    uploadFileError,
+}: Props) {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+
+    const [channelTimezoneCount, setChannelTimezoneCount] = useState(0);
+    const [sendingMessage, setSendingMessage] = useState(false);
+    const [channelMemberCountsByGroup, setChannelMemberCountsByGroup] = useState<ChannelMemberCountByGroup[]>([]);
+
+    const canSend = useCallback(() => {
+        if (sendingMessage) {
+            return false;
+        }
+
+        const messageLength = value.trim().length;
+
+        if (messageLength > maxMessageLength) {
+            return false;
+        }
+
+        if (files.length) {
+            const loadingComplete = !files.some((file) => file.loading);
+            return loadingComplete;
+        }
+
+        return messageLength > 0;
+    }, [sendingMessage, value, files, maxMessageLength]);
+
+    const sendReaction = useCallback((emoji: string) => {
+        addReactionToLatestPost(serverUrl, emoji, rootId);
+        clearDraft();
+        setSendingMessage(false);
+    }, [serverUrl, rootId, clearDraft]);
+
+    const doSubmitMessage = useCallback(() => {
+        const postFiles = files.filter((f) => !f.failed);
+        const post = {
+            user_id: currentUserId,
+            channel_id: channelId,
+            root_id: rootId,
+            message: value,
+        };
+
+        createPost(serverUrl, post, postFiles).then(({data}) => {
+            if (data) {
+                addRecentReaction(serverUrl, value);
+            }
+        });
+
+        clearDraft();
+        setSendingMessage(false);
+
+        // if (Platform.OS === 'android') {
```

Comment:
Are we keeping those commented lines for another PR ? if a todo, please add a ticket number